includes/classes/SupportMonitor/Monitor.php

```diff
@@ -388,6 +388,44 @@ public function send_message( $data, $type = 'notice', $group = 'message' ) {
 		return true;
 	}
 
+	/**
+	 * Check if loggin is enabled
+	 *
+	 * @return boolean
+	 */
+	public function logging_enabled() {
+		return ( ! defined( 'TENUP_DISABLE_ACTIVITYLOG' ) || ! TENUP_DISABLE_ACTIVITYLOG );
+	}
+
+	/**
+	 * Create a log entry
+	 *
+	 * @param string $summary  Data to log
+	 * @param string $subgroup Sub group
+	 */
+	public function log( $summary, $subgroup = null ) {
+		if ( ! $this->logging_enabled() ) {
+			return;
+		}
+
+		$current_logs = get_option( 'tenup_support_monitor_activity_logs', [] );
+
+		if ( apply_filters( 'tenup_support_monitor_max_activity_log_count', 100 ) <= count( $current_logs ) ) {
+			return;
+		}
+
+		$log_item = [
+			'date'     => time(),
+			'summary'  => $summary,
+			'subgroup' => $subgroup,
+			'user_id'  => get_current_user_id(),
+		];
+
+		$current_logs[] = $log_item;
+
+		update_option( 'tenup_support_monitor_activity_logs', $current_logs );
```

Comment:
@tlovett1 I'd suggest to add `false` to the autoload parameter in `update_option()` since there won't be a need for this data to be autoloaded into memory or caching. Leaving as true has the potential to also fill up a site's object caching with unnecessary data which usually only has a 1MB limit.